### ISSUE TITLE
File appender memory leak after reload configure

### ISSUE THREAD

**Issue by Cap32** [log4js-node/log4js-node#700]
I want to reload / update my configuration, but I get memory leak warning.

- log4js v2.5.3
- node.js v8.9.0
- macOS v10.13.4

```js
const log4js = require('log4js');
const options = {
	appenders: {
		foo: { type: 'file', filename: 'foo.log' },
	},
	categories: {
		default: { appenders: ['foo'], level: 'info' },
	},
};
const loop = (index = 0) => {
	if (index < 11) {
		log4js.configure(options);
		log4js.shutdown(() => loop(++index));
	}
};
loop();
```

throw:

`(node:19311) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 SIGHUP listeners added. Use emitter.setMaxListeners() to increase limit`

Or is there any other way to reload / update configuration?

Thanks

**Comment by nomiddlename** [comment]
Thanks for raising this - the file appender has a SIGHUP listener leak by the looks of things.

**Comment by nomiddlename** [comment]
Github auto-closed this, but the fix won't be released until version 2.6.0 goes out to NPM - I'm just waiting for a fix for a streamroller bug before I publish.

**Comment by lmyzzu** [comment]
when appender has too much(>=12), It's will emit this error

**Comment by yiu31802** [comment]
This issue still exists with the latest version 3.0.6. It started to appear when the number of appender reached 12.

**Comment by shudingbo** [comment]
This issue still exists with the latest version 3.0.6. It started to appear when the number of appender reached 12.

**Comment by peteriman** [comment]
This has been patched in PR #1110.
